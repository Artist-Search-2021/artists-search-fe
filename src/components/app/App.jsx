/* eslint-disable max-len */
import React from 'react';
import { Switch, Redirect, Route } from 'react-router-dom';
import ArtistReleases from '../display/ArtistReleases';
import ReleaseDetail from '../display/ReleaseDetail';
import SongLyrics from '../display/SongLyrics';
import Header from '../common/Header';
import Home from '../home/Home';

export default function App() {
  return (<div className="App">
   
    <Header />
    <Switch>
      <Route exact path="/" component={Home} />
      <Route exact path="/artist/:id" component={ArtistReleases}></Route>
      <Route exact path="/release/:id" component={ReleaseDetail}></Route>
      <Route exact path="/release/lyrics/:id" component={SongLyrics}></Route>
      <Redirect to="/" />
    </Switch>
    

  </div>);
}
