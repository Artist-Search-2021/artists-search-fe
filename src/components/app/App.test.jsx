import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import App from './App';
import { MemoryRouter } from 'react-router-dom';


jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), 
  useLocation: () => ({
        
        
    'pathname': '/release/21e8c81d-6d12-48a9-ad47-b371ca2c166b',
    'search': '',
    'hash': '',
    'state': {
      'artistName': {
        'pathname': '/artist/0743b15a-3c32-48c8-ad58-cb325350befa',
        'state': {
          'artistName': 'blink‐182'
        },
        'search': '',
        'hash': '',
        'key': 'fzjh2b'
      }
    },
    'key': '97cgva'
      
  })
}));

describe('artistSearch', () => {
  it('displays a list of artists', async ()  => {
    render(
      <MemoryRouter  initialEntries={['/']}>
        <App />
      </MemoryRouter>
    ); 
    // const { asFragment } = render(
    //   <Search/>
    // );

    // expect(asFragment()).toMatchSnapshot();
    
    screen.getByText('Search artists');
    const input = screen.getByPlaceholderText('Search artist');
    fireEvent.change(input, { target: { value: 'Prince' } });  
    expect(input.value).toBe('Prince');
    fireEvent.submit(input);
    const list = await screen.findByRole('list');
    expect(list).not.toBeEmptyDOMElement();
    expect(list.children.length).toEqual(25);
 
  });
  it('displays a list of recordings', async ()  => {
    render(
      <MemoryRouter  initialEntries={['/artist/070d193a-845c-479f-980e-bef15710653e']}>
        <App />
      </MemoryRouter>
    ); 
    const list = await screen.findByRole('list');
    expect(list).not.toBeEmptyDOMElement();
    expect(list.children.length).toEqual(12);

    
  });

  it('displays a list of song titles', async () => {
    

    render(
      <MemoryRouter initialEntries={['/release/21e8c81d-6d12-48a9-ad47-b371ca2c166b']}>
        <App />
      </MemoryRouter>
    );
   
    setTimeout(async () => {
      const list = await screen.findByRole('list');
      expect(list).not.toBeEmptyDOMElement();
      expect(list.children.length).toEqual(8);     
    }, 1000);
  });

 
});

// {
//   'state': {
//     'artistName': {
//       'pathname': '/artist/0743b15a-3c32-48c8-ad58-cb325350befa',
//       'state': {
//         'artistName': 'blink‐182'
//       },
//       'search': '',
//       'hash': '',
//       'key': 'fzjh2b'
//     }
//   }
// }
