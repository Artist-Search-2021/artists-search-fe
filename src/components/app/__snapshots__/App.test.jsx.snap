// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`artistSearch displays a list of artists 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        name="input"
        placeholder="Search artist"
        type="text"
        value=""
      />
    </form>
  </div>
</DocumentFragment>
`;
